src/stores/utilities/run.js

```diff
@@ -188,16 +190,11 @@ const hasPlateAttributes = ({ sequencing_kit_box_barcode, wells_attributes }) =>
  * @param {smrtLinkVersion} - The SMRT Link Version of the run
  * @returns {run} - A run object
  */
-const SMRT_LINK_VERSION_V12_SEQUEL_IIE = 'v12_sequel_iie'
-const SMRT_LINK_VERSION_V12_REVIO = 'v12_revio'
-
-const createRunPayload = (run, smrtLinkVersion) => {
-  // If the run is v12 Sequel IIe, remove redundant dna_control_complex_box_barcode
-  // dna_control_complex_box_barcode is currently an optional field in the service
-  // as it is not required for v12 Sequel IIe, but it for the other versions
-  if (
-    [SMRT_LINK_VERSION_V12_SEQUEL_IIE, SMRT_LINK_VERSION_V12_REVIO].includes(smrtLinkVersion.name)
-  ) {
+
+const createRunPayload = (run, instrumentType) => {
+  // If the system is Revio, remove redundant dna_control_complex_box_barcode
+  // as it is not required for Revio, but required for the other systems
+  if (instrumentType.key === PacbioInstrumentTypes.Revio.key) {
```

Comment:
Do we need to do this? If it is a Revio it is already null

Reply:
Yes. but, I thought this method is to make sure that  no unintended attributes are set for a smrtLinkVersion which was the case before, now changed to instrumentType. If we edit a SequleII run and changed the instrumentType to Revio

Reply:
Good thought. If you move it to the setInstrumentData method that would probably be better. That is where we remove all of the data if the instrument type is changed. https://github.com/sanger/traction-ui/blob/develop/src/stores/pacbioRunCreate.js#L432C29-L432C29

Reply:
That sounds a better place. I will check it out 👍  Thank you